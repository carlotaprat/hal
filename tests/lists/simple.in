list = [a, b, c, d, 3 + 2, f a, b, c]
