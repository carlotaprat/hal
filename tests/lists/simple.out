digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="="];
  n1 [label="="];
  n2 [label="FUNCALL"];
  n2 [label="FUNCALL"];
  n3 [label="list"];
  n4 [label="ARGS"];
  n5 [label="LIST"];
  n5 [label="LIST"];
  n6 [label="FUNCALL"];
  n6 [label="FUNCALL"];
  n7 [label="a"];
  n8 [label="ARGS"];
  n9 [label="FUNCALL"];
  n9 [label="FUNCALL"];
  n10 [label="b"];
  n11 [label="ARGS"];
  n12 [label="FUNCALL"];
  n12 [label="FUNCALL"];
  n13 [label="c"];
  n14 [label="ARGS"];
  n15 [label="FUNCALL"];
  n15 [label="FUNCALL"];
  n16 [label="d"];
  n17 [label="ARGS"];
  n18 [label="+"];
  n18 [label="+"];
  n19 [label="3"];
  n20 [label="2"];
  n21 [label="FUNCALL"];
  n21 [label="FUNCALL"];
  n22 [label="f"];
  n23 [label="ARGS"];
  n23 [label="ARGS"];
  n24 [label="FUNCALL"];
  n24 [label="FUNCALL"];
  n25 [label="a"];
  n26 [label="ARGS"];
  n27 [label="FUNCALL"];
  n27 [label="FUNCALL"];
  n28 [label="b"];
  n29 [label="ARGS"];
  n30 [label="FUNCALL"];
  n30 [label="FUNCALL"];
  n31 [label="c"];
  n32 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "="
  n1 -> n2 // "=" -> "FUNCALL"
  n2 -> n3 // "FUNCALL" -> "list"
  n2 -> n4 // "FUNCALL" -> "ARGS"
  n1 -> n5 // "=" -> "LIST"
  n5 -> n6 // "LIST" -> "FUNCALL"
  n6 -> n7 // "FUNCALL" -> "a"
  n6 -> n8 // "FUNCALL" -> "ARGS"
  n5 -> n9 // "LIST" -> "FUNCALL"
  n9 -> n10 // "FUNCALL" -> "b"
  n9 -> n11 // "FUNCALL" -> "ARGS"
  n5 -> n12 // "LIST" -> "FUNCALL"
  n12 -> n13 // "FUNCALL" -> "c"
  n12 -> n14 // "FUNCALL" -> "ARGS"
  n5 -> n15 // "LIST" -> "FUNCALL"
  n15 -> n16 // "FUNCALL" -> "d"
  n15 -> n17 // "FUNCALL" -> "ARGS"
  n5 -> n18 // "LIST" -> "+"
  n18 -> n19 // "+" -> "3"
  n18 -> n20 // "+" -> "2"
  n5 -> n21 // "LIST" -> "FUNCALL"
  n21 -> n22 // "FUNCALL" -> "f"
  n21 -> n23 // "FUNCALL" -> "ARGS"
  n23 -> n24 // "ARGS" -> "FUNCALL"
  n24 -> n25 // "FUNCALL" -> "a"
  n24 -> n26 // "FUNCALL" -> "ARGS"
  n23 -> n27 // "ARGS" -> "FUNCALL"
  n27 -> n28 // "FUNCALL" -> "b"
  n27 -> n29 // "FUNCALL" -> "ARGS"
  n23 -> n30 // "ARGS" -> "FUNCALL"
  n30 -> n31 // "FUNCALL" -> "c"
  n30 -> n32 // "FUNCALL" -> "ARGS"

}

